lib/math.py

```diff
@@ -0,0 +1,204 @@
+#NOTE: This is ugly.
+from __go__.math import (Pi, E, Ceil, Copysign, Abs, Floor, Mod, Frexp, IsInf,
+    IsNaN, Exp2, Modf, Trunc, Exp, Expm1, Log, Log1p, Log10, Pow, Sqrt, Acos,
+    Asin, Atan, Atan2, Hypot, Sin, Cos, Tan, Acosh, Asinh, Atanh, Sinh, Cosh,
+    Tanh, Erf, Erfc, Gamma, Lgamma) # pylint: disable=g-multiple-import
+
+# Constants
+
+pi = Pi
+
+
+e = E
+
+
+# Number-theoretic and representation functions
+
+def ceil(x):
+    return Ceil(float(x))
+    
+    
+def copysign(x,y):
+    return Copysign(float(x),float(y))
+
+    
+def fabs(x):
+    return Abs(float(x))
+
+
+def factorial(x):
+    
+    def factorial_helper(x, acc):
+        if x <= 1:
+            return acc
+        return factorial_helper(x - 1, acc * x)
+    
+    if x % 1 != 0 or x < 1:
+        raise ValueError
+
+    return factorial_helper(int(x),1)
```

Comment:
According to [the docs](https://docs.python.org/2/library/math.html#math.factorial), this should raise ValueError if `x` is not integral. Also, I'm not sure how much the exception messages matter, but here are the exact messages raised by `python2.7`:

```py
>>> math.factorial(2.1)
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
ValueError: factorial() only accepts integral values
>>> math.factorial(-1)
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
ValueError: factorial() not defined for negative values
```